app/services/order_tax_adjustments_fetcher.rb

```diff
@@ -20,24 +20,11 @@ def totals
   attr_reader :order
 
   def all
-    Spree::Adjustment
-      .where(order_adjustments.or(line_item_adjustments).or(shipment_adjustments))
-      .order('created_at ASC')
-  end
-
-  def order_adjustments
-    table[:adjustable_id].eq(order.id)
-      .and(table[:adjustable_type].eq('Spree::Order'))
-  end
-
-  def line_item_adjustments
-    table[:adjustable_id].eq_any(order.line_item_ids)
-      .and(table[:adjustable_type].eq('Spree::LineItem'))
-  end
+    tax_adjustments = order.all_adjustments.tax
+    enterprise_fees_with_tax = order.all_adjustments.enterprise_fee.with_tax
+    admin_adjustments_with_tax = order.adjustments.admin.with_tax
 
-  def shipment_adjustments
-    table[:order_id].eq(order.id)
-      .and(table[:adjustable_type].eq('Spree::Shipment'))
+    tax_adjustments | enterprise_fees_with_tax | admin_adjustments_with_tax
```

Comment:
Rails doesn't combine this as AR relation, is it?

Would this work?
```suggestion
    tax_adjustments.or(enterprise_fees_with_tax).or(admin_adjustments_with_tax)
```

Then we could add `.order('created_at ASC')` if needed as well. Did you drop that because it doesn't matter?

Reply:
Aha, yeah I wrote this before Rails 5 was merged. We can do nicer things with `or` now :tada:

I'll update it.

Reply:
Great, it works :100:

Commit added.